cvl/cvl.go

```diff
@@ -986,45 +1086,53 @@ func (c *CVL) doCustomValidation(node *xmlquery.Node,
 
 	// yangListName provides the correct table name defined in sonic-yang
 	// For ex. VLAN_INTERFACE_LIST and VLAN_INTERFACE_IPADDR_LIST are in same container
-	for nodeName, custFunc := range modelInfo.tableInfo[yangListName].custValidation {
+	for nodeName, custFuncs := range modelInfo.tableInfo[yangListName].custValidation {
 		//find the node value
 		//node value is empty for custom validation function at list level
+		//If key does not exist, then no need for custom validation
 		nodeVal := ""
+		fieldFound := false
 		if !strings.HasSuffix(nodeName, "_LIST") {
 			for nodeLeaf := node.FirstChild; nodeLeaf != nil; nodeLeaf = nodeLeaf.NextSibling {
 				if nodeName != nodeLeaf.Data {
 					continue
 				}
-
 				if (len(nodeLeaf.Attr) > 0) &&
 					(nodeLeaf.Attr[0].Name.Local == "leaf-list") {
 					nodeVal = curCustvCfg.Data[nodeName]
 				} else {
 					nodeVal = nodeLeaf.FirstChild.Data
 				}
+				fieldFound = true
+				break
+			}
+			if !fieldFound {
+				CVL_LOG(INFO_DEBUG, "Non list with empty entry, no need of custom validation for %s", nodeName)
+				continue
 			}
 
 		}
 
 		//Call custom validation functions
-		CVL_LOG(INFO_TRACE, "Calling custom validation function %s", custFunc)
-		pCustv := &custv.CustValidationCtxt{
-			ReqData:   custvCfg,
-			CurCfg:    curCustvCfg,
-			YNodeName: nodeName,
-			YNodeVal:  nodeVal,
-			YCur:      node,
-			SessCache: &(c.custvCache),
-			RClient:   redisClient}
-
-		errObj := custv.InvokeCustomValidation(&custv.CustomValidation{},
-			custFunc, pCustv)
-
-		cvlErrObj = *(*CVLErrorInfo)(unsafe.Pointer(&errObj))
-
-		if cvlErrObj.ErrCode != CVL_SUCCESS {
-			CVL_LOG(WARNING, "Custom validation failed, Error = %v", cvlErrObj)
-			return cvlErrObj
+		for _, custFunction := range custFuncs {
+			CVL_LOG(INFO_TRACE, "Calling custom validation function %s", custFunction)
+			pCustv := &custv.CustValidationCtxt{
```

Comment:
Move the creation of pCustv outside the loop: Creating pCustv inside the loop can result in unnecessary object creation and garbage collection. You can move the creation of pCustv outside the loop to avoid this overhead.